components/brave_wallet_ui/components/desktop/views/portfolio/components/nft-details/index.tsx

```diff
@@ -85,61 +87,80 @@ const NFTDetails = (props: Props) => {
 
   return (
     <StyledWrapper>
-      <NFTImage src={selectedAsset.logo} />
-      <DetailColumn>
-        <TokenName>
-          {selectedAsset.name} {
-            selectedAsset.tokenId
-              ? '#' + new Amount(selectedAsset.tokenId).toNumber()
-              : ''
+      {isLoading &&
+        <LoadingOverlay isLoading={isLoading}>
+          <LoadIcon />
+        </LoadingOverlay>
+      }
+      {nftMetadata &&
+        <>
+          <NTFImageIframe
```

Comment:
why are we loading each NFT as its own chrome-untrusted frame instead of loading all the NFTs in a single chrome-untrusted frame? cc @petemill

Reply:
The `chrome-untrusted` page receives an imageUrl as a query parameter and displays a single image. Passing multiple images as query parameters will possible hit the limit for query params. Also, changing the page structure to display all images would adding the UI(table) in the chrome-untrusted tab. Are there security or performance issues with the current implementation?

Reply:
how many iframes do we expect to create on a single view?

Reply:
@petemill An iframe for each image.

Reply:
that seems like it could be a performance hit... security-wise, the benefit of doing so is to isolate NFTs from each other if one NFT image is malicious. however i'm not sure this is worth it. the main point of doing the chrome-untrusted isolation is to isolate the main `chrome://wallet` context (which has access to private keys) from a process where untrusted code can run, not isolate the untrusted code from each other. will defer to @petemill though

Reply:
@petemill Please add your thoughts on this.

Reply:
I'm thinking this change, might actually be useful to avoid the possibility that one NFT could be loading JS from an SVG to modify the img-src of a second NFT to make it appear as if it's a more valuable NFT and cause a discrepancy between what's rendered in our wallet versus others.